File: framework/languages/dotnet/e2e/LocationMessaging/Client/Scenarios/RmA2ManualEndpointScenario.cs
// Verifies RM-A2 Manual Endpoint behavior.
using LocationMessaging.Client.Support;
using LocationMessaging.Shared;
using Zlink.HttpClient;

namespace LocationMessaging.Client.Scenarios;

// RM-A2 verifies that a consumer can request a profile provider through an
// explicitly configured peer endpoint without registering a location store.
internal static class RmA2ManualEndpointScenario
{
    public static async Task RunAsync(
        ZLinkHttpClient singleConsumer,
        ZLinkHttpClient providerA)
    {
        var reply = (await singleConsumer.Post("/profile/request").Body(new ProfileReq("rm-a2"))
            .Async<ProfileRes>()).Body;
        ZlinkStreamAssert.Ensure(reply.Value == "profile:rm-a2", "RM-A2 reply value mismatch.");
        ZlinkStreamAssert.Ensure(reply.ProviderRid == "api-a", "RM-A2 manual endpoint should reach api-a.");

        var evidence = (await providerA.Post("/evidence/wait")
            .Body(new EvidenceWaitReq("value=rm-a2"))
            .Async<string[]>()).Body;
        ZlinkStreamAssert.Ensure(
            evidence.Any(line => line.Contains("value=rm-a2", StringComparison.Ordinal)),
            "RM-A2 api-a evidence missing.");
    }
}

File: framework/languages/java/e2e-kotlin/RegistryMessaging/Client/src/main/kotlin/systems/zlink/e2e/kotlin/registrymessaging/client/Scenarios/RmA2ManualEndpointScenario.kt
package systems.zlink.e2e.kotlin.registrymessaging.client.Scenarios

import systems.zlink.e2e.kotlin.registrymessaging.client.Support.HttpJson
import systems.zlink.e2e.kotlin.registrymessaging.client.Support.ScenarioAssert
import systems.zlink.e2e.kotlin.registrymessaging.shared.EvidenceWaitReq
import systems.zlink.e2e.kotlin.registrymessaging.shared.ProfileRes
import systems.zlink.e2e.kotlin.registrymessaging.shared.ProfileReq

object RmA2ManualEndpointScenario {
    fun run(singleConsumer: HttpJson, providerA: HttpJson) {
        val reply = singleConsumer.post<ProfileRes>("/profile/request", ProfileReq("rm-a2"))
        ScenarioAssert.that(reply.value == "profile:rm-a2", "RM-A2 reply value mismatch.")
        ScenarioAssert.that(reply.providerRid == "api-a", "RM-A2 manual endpoint should reach api-a.")
        val evidence = providerA.post<List<String>>("/evidence/wait", EvidenceWaitReq("value=rm-a2"))
        ScenarioAssert.that(evidence.any { it.contains("value=rm-a2") }, "RM-A2 api-a evidence missing.")
        println("scenario RM-A2 passed")
    }
}
